api/types/types.go

```diff
@@ -202,22 +202,29 @@ type Info struct {
 	// LiveRestoreEnabled determines whether containers should be kept
 	// running when the daemon is shutdown or upon daemon start if
 	// running containers are detected
-	LiveRestoreEnabled bool
-	Isolation          container.Isolation
-	InitBinary         string
-	ContainerdCommit   Commit
-	RuncCommit         Commit
-	InitCommit         Commit
-	SecurityOptions    []string
-	ProductLicense     string `json:",omitempty"`
-	Warnings           []string
+	LiveRestoreEnabled  bool
+	Isolation           container.Isolation
+	InitBinary          string
+	ContainerdCommit    Commit
+	RuncCommit          Commit
+	InitCommit          Commit
+	SecurityOptions     []string
+	ProductLicense      string `json:",omitempty"`
+	DefaultAddressPools []NetworkAddressPool
+	Warnings            []string
 }
 
 // KeyValue holds a key/value pair
 type KeyValue struct {
 	Key, Value string
 }
 
+// NetworkAddressPool is a temp struct used by Info struct
+type NetworkAddressPool struct {
+	Base string
+	Size int
+}
```

Comment:
I actually just realised that for the SwarmKit default-addresspools, we use a regular `[]string`;

```
Default Address Pool: 10.0.0.0/8
```

```json
"DefaultAddrPool": [
"10.0.0.0/8"
],
```

Should we do the same here?

Reply:
I think `DefaultAddrPool` of SwarmKit is different with `DefaultAddressPools` here. `DefaultAddressPools` contains `CIDR` and pool `Size` but `DefaultAddrPool` only contains `CIDR`, so we can not use regular `[]string` . 

```
"default-address-pools":  [
 { "base": "10.123.0.0/16", "size": 24 },
 { "base": "10.12.0.0/16", "size": 24 } 
]
```

Reply:
Ah, yes, I recall now that we had some back-and-forth discussion about them being different 👍